### ISSUE TITLE
Dropped functionality: Ability to set "empty text" for UI Component listings

### ISSUE THREAD

**Issue by Pinchanzee** [magento/magento2#33473]
In Magento1 it was possible to change the text displayed in the dummy empty grid row, by setting _emptyText.
In Magento2 this is hardcoded for UI Components to "We couldn't find any records." in /module-ui/view/base/web/templates/grid/listing.html

Would it be difficult to implement this as a feature of the component's XML?

**Comment by engcom-Delta** [comment]
Hi @Pinchanzee,
Kindly provide the steps to reproduce the issue, Expected results and Actual results.

Thanks

**Comment by Pinchanzee** [comment]
This is a feature request, reproduction steps are not applicable

**Comment by mrtuvn** [comment]
wonder why do you not replace in custom module or custom theme in admin

**Comment by Pinchanzee** [comment]
Yes I could do this, but I'm not really that passionate about it for my
particular use case.

I just think it's a useful feature to have dropped in the migration to UI
Components that likely wasn't recorded anywhere.

**Comment by engcom-Delta** [comment]
Hi @Pinchanzee,

Kindly provide use cases for this feature request.

Thanks

**Comment by Pinchanzee** [comment]
Any time a grid is empty, it can be helpful to provide the user with more feedback.
1. Cases where a grid is not being used as a record searching tool.  Eg Grid of ERP picking sessions that when empty says "Nothing is currently being picked right now" rather than "We couldn't find any records" - the latter is confusing to the end user when nothing is being searched for.
2. Where you want to provide helpful information for the end user when there are no results.  Eg. "No results, last synced [date]".  Or "No results.  If you are instead looking for [X] be sure to check [Y]"
